lightning/src/util/persist.rs

```diff
@@ -491,14 +499,45 @@ where
 	/// less frequent "waves."
 	///   - [`MonitorUpdatingPersister`] will potentially have more listing to do if you need to run
 	/// [`MonitorUpdatingPersister::cleanup_stale_updates`].
+	///
+	/// This sets `min_monitor_size_for_updates_bytes` to 8192 bytes (8 KiB), which is a reasonable
+	/// default for most use cases. Monitors smaller than this will be persisted in full rather than
+	/// using update-based persistence. Use [`MonitorUpdatingPersister::new_with_monitor_size_threshold`]
+	/// if you need a custom threshold.
 	pub fn new(
 		kv_store: K, logger: L, maximum_pending_updates: u64, entropy_source: ES,
 		signer_provider: SP, broadcaster: BI, fee_estimator: FE,
+	) -> Self {
+		Self::new_with_monitor_size_threshold(
+			kv_store,
+			logger,
+			maximum_pending_updates,
+			8192,
+			entropy_source,
+			signer_provider,
+			broadcaster,
+			fee_estimator,
+		)
+	}
+
+	/// Constructs a new [`MonitorUpdatingPersister`] with a custom minimum monitor size threshold.
+	///
+	/// The `min_monitor_size_for_updates_bytes` parameter sets the minimum serialized size (in bytes)
+	/// for a [`ChannelMonitor`] to use update-based persistence. Monitors smaller than this threshold
+	/// will always be persisted in full, avoiding the overhead of managing update files for tiny
+	/// monitors. Set to 0 to always use update-based persistence regardless of size.
+	///
+	/// For other parameters, see [`MonitorUpdatingPersister::new`].
+	pub fn new_with_monitor_size_threshold(
```

Comment:
I wonder if, instead of two constructors, it would be nicer/better to either add this to the default constructor, or use a config struct, or write a builder. 

It's not a problem here, but I wonder if we'll think of another tunable for MUP and then there will be a third constructor, and so on.